src/tests/integration_tests/test_qa_agent_with_vector_shared_memory.py

```diff
@@ -0,0 +1,75 @@
+Human teeth are the only part of the body that cannot heal themselves. Teeth are coated in enamel which is not a living tissue.
+It's illegal to own just one guinea pig in Switzerland. It's considered animal abuse because they're social beings and get lonely.
+The Ancient Romans used to drop a piece of toast into their wine for good health - hence why we 'raise a toast'.
+The heart of a shrimp is located in its head. They also have an open circulatory system, which means they have no arteries and their organs float directly in blood.
+Amy Poehler was only seven years older than Rachel McAdams when she took on the role of "cool mom" in Mean Girls. Rachel was 25 as Regina George - Amy was 32 as her mum.
+People are more creative in the shower. When we take a warm shower, we experience an increased dopamine flow that makes us more creative.
+Baby rabbits are called kits. Cute!
+my dog died on march 2021.
+The unicorn is the national animal of Scotland. It was apparently chosen because of its connection with dominance and chivalry as well as purity and innocence in Celtic mythology.
+The first aeroplane flew on December 17,1903 and  . Wilbur and Orville Wright made four brief flights at Kitty Hawk, North Carolina, with their first powered aircraft, aka the first airplane.
+Venus is the only planet to spin clockwise. It travels around the sun once every 225 Earth days but it rotates clockwise once every 243 days.
+Nutmeg is a hallucinogen. The spice contains myristicin, a natural compound that has mind-altering effects if ingested in large doses.
+A 73-year-old bottle of French Burgundy became the most expensive bottle of wine ever sold at auction in 2018, going for $558,000 (approx £439,300). The bottle of 1945 Romanee-Conti sold at Sotheby for more than 17 times its original estimate of $32,000."""
+session_id = "6"
+meta_data = {
+    "session_id": f"{session_id}",
+    "file_name": "rtgfqq",
+    "file_type": "pdf",
+    "title": "NoMeaning",
+    "data_type": "user_input",
+}
+
+
+@pytest.mark.external_api
+def test_shared_memory_with_vector(get_llm):
+    llm = get_llm(__file__, test_shared_memory_with_vector.__name__)
+    # store text as a scraped text from a file with meta_data session_id
+    split_data = ChromaVectorStore.file_text_splitter(data=data, meta_data=meta_data)
+    ChromaVectorStore.chroma_from_texts(
+        texts=split_data["texts"], meta_datas=split_data["meta_datas"]
+    )
+
+    shared_memory = SharedMemoryWithVectorDB(
+        objective="summerize the file rtgfqq.",
+        agent_pool=None,
+        session_id=session_id,
+    )
+
+    task_agent = QAAgent(
+        llm=llm,
+        shared_memory=shared_memory,
+    )
+
+    shared_memory.add(
+        EventType.task,
+        "Planner",
+        "summerize the file rtgfqq",
+    )
+
+    task_agent.run()
+
+    results = shared_memory.get_by_type(EventType.result)
+    print(results[0].content)
+    logger.debug(results[0].content)
+    assert len(results) == 1
```

Comment:
Is there a way to also check the vector db is actually working? Like checking the file, check that the vectordb gets the correct document from the mock data.

Reply:
I just created a test for the similarity and also check if the result is from the desired chunk